site-src/geps/gep-724.md

```diff
@@ -0,0 +1,614 @@
+
+  // GatewayRefs must be specified when Allow is set to "FromList". In that
+  // case, only Gateways referenced in this list will be allowed to use this
+  // route. This field is ignored for other values of "Allow".
+  //
+  // +optional
+  GatewayRefs []GatewayReference `json:"gatewayRefs,omitempty"`
+}
+```
+
+#### Added
+To Route Specs:
+```go
+  // AttachTo defines the parent resources this Route should be attached to. The
+  // only kind of target resource with "Core" support is Gateway. This API may
+  // be extended in the future to support additional kinds of parent resources
+  // such as Routes.
+  //
+  // +optional
+  // +kubebuilder:validation:MaxItems=16
+  AttachTo []AttachRef `json:"gateways,omitempty"`
+```
+
+And the struct that references:
+```go
+// AttachRef identifies an API object to attach to. The only kind of target
+// resource with "Core" support is Gateway. This API may be extended in the
+// future to support additional kinds of parent resources such as Routes.
+type AttachRef struct {
+  // Group is the group of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=gateway.networking.k8s.io
+  Group string `json:"group"`
+
+  // Kind is kind of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Kind string `json:"kind"`
+
+  // Namespace is the namespace of the referent. When unspecified (empty
+  // string), this will either be:
+  //
+  // * local namespace of the target is a namespace scoped resource
+  // * no namespace (not applicable) if the target is cluster-scoped.
+  //
+  // Support: Extended
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  Namespace string `json:"namespace,omitempty"`
+
+  // Name is the name of the referent.
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  Name string `json:"name"`
+
+  // SectionName is the name of a section within the target resource. In the
+  // following resources, SectionName is interpreted as the following:
+  //
+  // * Gateway: Listener Name
+  //
+  // Implementations MAY choose to support attaching Routes to other resources.
+  // If that is the case, they MUST clearly document how SectionName is
+  // interpreted.
+  //
+  // When unspecified (empty string), this will reference the entire resource.
+  // For the purpose of status, an attachment is considered successful if at
+  // least one section in the parent resource accepts it. For example, Gateway
+  // listeners can restrict which Routes can bind to them by Route kind,
+  // namespace, or hostname. If 1 of 2 Gateway listeners accept attachment from
+  // the referencing Route, the Route MUST be considered successfully
+  // attached. If no Gateway listeners accept attachment from this Route, the
+  // Route MUST be considered detached from the Gateway.
```

Comment:
Did we consider a []string for this type to support explicitly tartgetting ports? Not a requirement?

Reply:
I think it's always easier to go from a single value to a list than the other way around (for example Pod IP -> IPs now). My preference is to start on the simpler side and expand if/when it's needed.